### PR TITLE
Updated CUDA Name Clash Checker By Added CUDA-specific keywords

### PR DESCRIPTION
**Pull request by smazouz42**
This pull request addresses issue #59  by adding  more CUDA-specific keywords to enhance the checking of variable/function names and prevent name clashes

### CODE REVIEW

File: pyccel/naming/cudanameclashchecker.py [hunk 1]
```diff
@@ -16,6 +16,10 @@ class CudaNameClashChecker(LanguageNameClashChecker):
     verify that they do not cause name clashes. Name clashes may be due to
     new variables, or due to the use of reserved keywords.
     """
+
+    def __init__(self) -> None:
+        super().__init__()
```
Comment (EmilyBourne) [inline pyccel/naming/cudanameclashchecker.py:21]: :| these warnings are a bit annoying on cases like this. Can you do the same as here:
https://github.com/pyccel/pyccel-cuda/blob/bd7351493e3ae2c0947b1d2fb92605360db4de08/pyccel/ast/datatypes.py#L66-L69

but in the superclass please

File: pyccel/naming/cudanameclashchecker.py [hunk 2]
```diff
@@ -16,6 +16,12 @@ class CudaNameClashChecker(LanguageNameClashChecker):
     verify that they do not cause name clashes. Name clashes may be due to
     new variables, or due to the use of reserved keywords.
     """
+
+    def __init__(self): #pylint: disable=useless-parent-delegation
+    # This __init__ function is required so the ArgumentSingleton can
+    # always detect a signature
+        super().__init__()
```
Comment (EmilyBourne) [inline pyccel/naming/cudanameclashchecker.py:23]: As requested in my last commet. Please put this constructor in the superclass

**Review by EmilyBourne — CHANGES_REQUESTED** [review]

File: pyccel/naming/languagenameclashchecker.py [hunk 3]
```diff
@@ -17,8 +17,12 @@ class LanguageNameClashChecker(metaclass = Singleton):
     verify that they do not cause name clashes. Name clashes may be due to
     a variety of reasons which vary from language to language.
     """
-    keywords = None
+    def __init__(self): #pylint: disable=useless-parent-delegation
```
Comment (EmilyBourne) [inline pyccel/naming/languagenameclashchecker.py:20]: Methods usually come after class variables

**Review by EmilyBourne — CHANGES_REQUESTED** [review]

File: pyccel/naming/languagenameclashchecker.py (hunk 3 above)
Comment (smazouz42) [reply pyccel/naming/languagenameclashchecker.py]: [ba81868](https://github.com/pyccel/pyccel-cuda/pull/60/commits/ba81868f6b47d37738a4148a3cd83c3647dac994)

**Review by EmilyBourne — APPROVED** [review]

**Comment by EmilyBourne** [comment]
Please fix merge conflicts so this can be merged

File: pyccel/naming/languagenameclashchecker.py [hunk 4]
```diff
@@ -19,6 +19,11 @@ class LanguageNameClashChecker(metaclass = Singleton):
     """
     keywords = None
 
+    def __init__(self): #pylint: disable=useless-parent-delegation
+    # This __init__ function is required so the ArgumentSingleton can
+    # always detect a signature
+        super().__init__()
```
Comment (EmilyBourne) [inline pyccel/naming/languagenameclashchecker.py:25]: For ease of parsing I would recommend putting the comment inside the function
```suggestion
    def __init__(self): #pylint: disable=useless-parent-delegation
        # This __init__ function is required so the ArgumentSingleton can
        # always detect a signature
        super().__init__()
```

**Review by EmilyBourne — CHANGES_REQUESTED** [review]

File: pyccel/naming/languagenameclashchecker.py (hunk 4 above)
Comment (smazouz42) [reply pyccel/naming/languagenameclashchecker.py]: [8cc3231](https://github.com/pyccel/pyccel-cuda/pull/60/commits/8cc32315d4c1a8020b2c3c1aaa0189d03ef82257)

**Review by EmilyBourne — APPROVED** [review]

**Pull request merged**